### PR TITLE
feat: allow consumers to configure ModalDialog to allow overflow

### PR DESCRIPTION
**Pull request by brian-smith-tcril**
## Description

Resolves https://github.com/openedx/paragon/issues/2938

### [Playground](https://deploy-preview-2939--paragon-openedx.netlify.app/playroom/?code=N4Igxg9gJgpiBcJgApkEoAEBeAfB4AOgHYYaREDOALhgNoCWFA8gA4xEA0GFMVr7AXWwYArjwDKVAIZUYyAGZSANjzQBuYqXLU6PJTDCyoXHlXEx9hmFCFZRE6bOQByZ%2Bs0YPAJ14ivJZA9SAB4cINIMYIAhESoqCBIANykveikiKiwCEAhYpXoiGABaFlSAWxSAT2yMBIBhfLAAaywUTFxuXn4iZCovERg0AF8wkgjx7owy6GVwkIB6GLiE0bnIgFkZpQARNKUIAHM10ip6Kn0skAAJGCkoClEWAEJs44xGbtaPtiIht-r9jxWuhsHhTN0FMpVH8xuNSB9EjAvPJ9gB3ABqjHoACN9GtRnDxsFNlBlLtlIcAHQ3O5IgmEiLErbk-YHSkAFTO%2BnpDIiJOUGFRZwAFhgAIKxCAUEQHA4wHRvBb8nZ7Kmc84wHlw4LzZUsqk02BeLVEvWqtlRaCVE2MgBiEC8ZUpEvi0tl8qoioiKIgMgKBwAMlJsRZLgAFLyHLxSMplf0YJTpA4iKRy16w3kYFJpIqJkNKS7yB1lLOSt1y6jpzMRYUWFjreUUVMwS7mSw0RNEZPNqvVjBIyNeBsUJty9kwAAemWyAFEvIOuEQIJ129YMMklANe9WNwNWnoDEYYX3akQ24frMDd4NQZ0zBYL1BkNfhm8bYTgvbHc6yzKK3wWFOFYACkpGScQwFSQCdS-J0XSlP8PVYICiHfbVYJ-V1EOoZD6BWMNKioYUEhg4tMIQ90cMAvDUK9EIMPg8skOolYACURBDUjv0Y7CAJQtCiQY39KL4mjTwaehmmBG8OmUJEqGQGBKTAPwfAydkUjlKhKS0sU4lSbFYjkZxSWkIprzcYYBIZXCEkFEUyDEeISwBSSmjoyJ5iErCRNs2iMyJLyyJ4kS0J1M0KQtK1rKZUkVUiyl7QgWRjS9YIxUMGjWIgVFrJCCLWUpBopRgJZ4iSbN0mnEAUtOKpsjyiI6nSMALA88LmXNIrAVK2Jysa6I%2Brs5JUiqy5SnoCovGqEBGtIMUMEM5YiHaxYhv86sdQylDstyt4Ori-U2SSlKbQOslzS1HV6XcX40HQIYQCGIA)

image

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [ ] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

File: src/Modal/_ModalDialog.scss [hunk 1]
```diff
@@ -38,6 +38,14 @@
...
     }
   }
+
+  &.pgn__modal-visible-overflow {
+    overflow: visible;
+
+    .pgn__modal-body {
+      overflow: visible;
+    }
+  }
```
Comment (PKulkoRaccoonGang) [inline src/Modal/_ModalDialog.scss:48]: Please try this. I think this option looks better:

```suggestion
  &.pgn__modal-visible-overflow .pgn__modal-body {
    overflow: visible;
  }
```

File: src/Modal/ModalDialog.jsx [hunk 2]
```diff
@@ -117,6 +119,7 @@ ModalDialog.propTypes = {
    */
   isBlocking: PropTypes.bool,
   zIndex: PropTypes.number,
+  isOverflowVisible: PropTypes.bool,
```
Comment (PKulkoRaccoonGang) [inline src/Modal/ModalDialog.jsx:122]: For all the properties of this component, we provide a short description for the convenience of consumers. I suggest doing this also for `zIndex` and `isOverflowVisible`

For example:
```
  /** Specifies the z-index of the modal */
  zIndex: PropTypes.number,
  /** Specifies whether overflow is visible in the modal */
  isOverflowVisible: PropTypes.bool,
```

File: src/Modal/_ModalDialog.scss (hunk 1 above)
Comment (brian-smith-tcril) [reply src/Modal/_ModalDialog.scss:48]: That doesn't work as we need to overflow both `pgn__modal-body` and `pgn__modal`. Your suggestion is missing the
```scss
  &.pgn__modal-visible-overflow {
    overflow: visible;
  }
```
part.

File: src/Modal/ModalDialog.jsx (hunk 2 above)
Comment (brian-smith-tcril) [reply src/Modal/ModalDialog.jsx]: Good catch! Addressed here https://github.com/openedx/paragon/compare/2f7f60c6a9ac8348c614ae9954ac9fd7a8a6c293..fc31ad7e3c3c69968d89712a15f9d42672ddabd6

**Review by PKulkoRaccoonGang — APPROVED** [review]

**Comment by rpenido** [comment]
Thank you for this PR @brian-smith-tcril! :+1: 
I'm using this CSS before we merge and update the package!

I found a nit: The top border of the dialog is not rounded after setting the overflow.
image

But if it's not easily fixed, I don't think this should prevent this PR from getting merged.

**Comment by brian-smith-tcril** [comment]
> Thank you for this PR @brian-smith-tcril! 👍 I'm using this CSS before we merge and update the package!
> 
> I found a nit: The top border of the dialog is not rounded after setting the overflow. image
> 
> But if it's not easily fixed, I don't think this should prevent this PR from getting merged.

Fixed! https://github.com/openedx/paragon/compare/6b611a0cdf4a65826a3ce48491ea173a35a3ce16..a1acfa0375f16088d49ea46a6f2acfa94a222bcf

File: src/Modal/ModalDialog.jsx [hunk 3]
```diff
@@ -133,6 +137,7 @@ ModalDialog.defaultProps = {
   isFullscreenOnMobile: false,
   isBlocking: false,
   zIndex: undefined,
+  isOverflowVisible: false,
```
Comment (brian-smith-tcril) [inline src/Modal/ModalDialog.jsx:140]: default to true

**Review by adamstankiewicz — APPROVED** [review]

**Pull request merged**